src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/resource/CsvDataExtractionController.kt

```diff
@@ -120,19 +90,20 @@ class CsvDataExtractionController(private val service: CsvDataExtractionService)
     @Parameter(description = "Return details of bookings occurring within this number of days of start-date")
     days: Long = 7,
   ): ResponseEntity<StreamingResponseBody> =
-    ResponseEntity.ok()
-      .header(
-        HttpHeaders.CONTENT_DISPOSITION,
-        "attachment; filename=video-links-by-probation-booking-date-from-${startDate.toIsoDate()}-for-$days-days.csv",
-      )
+    streamedCsvResponse(
+      startDate,
+      days,
+      "video-links-by-probation-booking-date-from-${startDate.toIsoDate()}-for-$days-days.csv",
+    ) { os -> service.probationBookingsByBookingDateToCsv(fromDate = startDate, toDate = startDate.plusDays(days), os) }
+
+  private fun streamedCsvResponse(startDate: LocalDate, days: Long, filename: String, block: (outputStream: OutputStream) -> Unit): ResponseEntity<StreamingResponseBody> {
+    require(ChronoUnit.DAYS.between(startDate, startDate.plusDays(days)) <= 365) {
```

Comment:
Brought check into here as 400 bad request was not being thrown, 500 internal error was being raised when the same check is being carried out in the service layer.

The controller advice is not able to interpret it as it is was being treated as text and not JSON due to the set up of the ResponseEntity below.